vispy/visuals/collections/agg_fast_path_collection.py

```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014, Nicolas P. Rougier
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+"""
+Antigrain Geometry Fast Path Collection
+
+This collection provides antialiased and accurate paths with caps and miter
+joins. It consume x4 more memory than regular lines and is a bit slower, but
+the quality of the output is worth the cost. Note that no control can be made
+on miter joins which may result in some glitches on screen.
+"""
+import numpy as np
+from vispy import glsl
+from vispy.gloo import gl
+from . collection import Collection
+
+
+class AggFastPathCollection(Collection):
+
+    """
+    Antigrain Geometry Fast Path Collection
+
+    This collection provides antialiased and accurate paths with caps and miter
+    joins. It consume x4 more memory than regular lines and is a bit slower,
+    but the quality of the output is worth the cost. Note that no control can
+    be made on miter joins which may result in some glitches on screen.
+    """
+
+    def __init__(self, user_dtype=None, transform=None,
+                 vertex=None, fragment=None, **kwargs):
+        """
+        Initialize the collection.
+
+        Parameters
+        ----------
+
+        user_dtype: list
+            The base dtype can be completed (appended) by the used_dtype. It
+            only make sense if user also provide vertex and/or fragment shaders
+
+        transform : string
+            GLSL Transform code defining the vec4 transform(vec3) function
+
+        vertex: string
+            Vertex shader code
+
+        fragment: string
+            Fragment  shader code
+
+        caps : string
+            'local', 'shared' or 'global'
+
+        color : string
+            'local', 'shared' or 'global'
+
+        linewidth : string
+            'local', 'shared' or 'global'
+
+        antialias : string
+            'local', 'shared' or 'global'
```

Comment:
In all our docs we don't have whitespace between parameters, so should probably be removed here. Also maybe we should explain what 'local', 'shared' and 'global' means (or refer to where its explained)?